Wikipedia/Code/WMFSearchViewController.m

```diff
@@ -501,6 +507,9 @@ - (void)updateLanguageButtonsToPreferredLanguages {
 - (void)resizeLanguageButtonsIfNeeded {
     [self.languageButtons enumerateObjectsUsingBlock:^(UIButton* _Nonnull obj, NSUInteger idx, BOOL* _Nonnull stop) {
         [obj sizeToFit];
+        CGRect f = obj.frame;
+        f.size.height = 30.0f;
+        obj.frame = f;
```

Comment:
[obj sizeToFit];
^ This changes the height from storyboard height. that's why the bug. first time you see these buttons, the height of button is 44px. (so the bottom line is sticking bottom of the toolbar) and when you change the language, height of the button is 30px so the line has a gap. we want to keep the height 30px. maybe this is a wrong way to do it. I will consult @montehurd on how to do this right. i saw it done like this in the same file so i copy-pasted the code.